src/endpoints/Collections.php

```diff
@@ -67,12 +68,17 @@ public function all(Request $request, Response $response)
      */
     public function read(Request $request, Response $response)
     {
+        $name = $request->getAttribute('name');
         $service = new TablesService($this->container);
         $responseData = $service->findByIds(
-            $request->getAttribute('name'),
+            $name,
             $request->getQueryParams()
         );
 
+        if(count($responseData['data']) === 0){
+            throw new CollectionNotFoundException($request->getAttribute($name));
```

Comment:
`$request->getAttribute($name)` will not get the expected value, as if I request `/collections/users`, it will try to get an attribute named `users` `$request->getAttribute('users')`

I am glad you are taking your time to get this PRs, but the collections and fields endpoint are more complex and it needs set the tables service where it can be shared across the codebase.

If you remove these changes I will merge it and we can discuss more about your ideas to solve this issue.